pkg/broker/filter/filter_handler.go

```diff
@@ -72,6 +72,11 @@ const (
 	skipTTL        = -1
 )
 
+var (
+	brokerNamespace string
+	brokerRef       string
+)
+
```

Comment:
I wouldn't make these variables global in the package, the various `handle*` functions could be called concurrently, so you will get weird results with these variables being global.

Instead, I would create them once inside each function